src/runtime/bootstrap-custom-element.ts

```diff
@@ -89,6 +98,19 @@ export const proxyCustomElement = (Cstr: any, compactMeta: d.ComponentRuntimeMet
       if (BUILD.disconnectedCallback && originalDisconnectedCallback) {
         originalDisconnectedCallback.call(this);
       }
+
+      /**
+       * Clean up `hostRefs` WeakMap when the element is disconnected
+       */
+      plt.raf(() => {
+        const hostRef = getHostRef(this);
+        if (hostRef?.$vnode$?.$elm$ instanceof Node && hostRef.$vnode$.$elm$.isConnected) {
```

Comment:
Hey @christian-bromann, I think it might need to be a negation of isConnected.
```suggestion
        if (hostRef?.$vnode$?.$elm$ instanceof Node && !hostRef.$vnode$.$elm$.isConnected) {
```